### PR TITLE
Add CLI command to verify ingestion by getting ad directly from provider

### PR DESCRIPTION
**Pull request by masih**
Add `provider verify-ingest` option to get the advertisement by directly
connecting to a provider's grphsync endpoint and fetching either the
latest advertisement entries or entries belonging to a specific
advertisement CID.

### CODE REVIEW

File: cmd/provider/doc.go [hunk 1]
```diff
@@ -24,6 +24,7 @@ Usage:
 	   register           Register provider information with an indexer that trusts the provider
 	   remove, rm         Removes previously advertised multihashes by the provider.
 	   verify-ingest, vi  Verifies ingestion of multihashes to an indexer node from a Lotus miner, CAR file or a CARv2 Index
+	   get-ad             Gets advertisement
```
Comment (willscott) [inline cmd/provider/doc.go:27]: maybe `list`?

File: cmd/provider/get.go [hunk 2]
```diff
@@ -62,41 +64,30 @@ func beforeGetAdvertisements(cctx *cli.Context) error {
...
-	} else {
-		if topic == "" {
-			return cli.Exit("topic must be configured when graphsync endpoint is specified.", 1)
-		}
-		provClient, err = internal.NewGraphSyncProviderClient(addrInfo, topic)
-		if err != nil {
-			return err
-		}
+	if topic == "" {
+		return nil, errors.New("topic must be configured when graphsync endpoint is specified")
```
Comment (willscott) [inline cmd/provider/get.go:88]: why is this? i thought topic was just for the gossiping - is it also used as the name of the `head` service?

Comment (masih) [reply cmd/provider/get.go:88]: Yes for the head.

Comment (willscott) [reply cmd/provider/get.go:88]: there is i believe a way to enumerate supported protocols with multistream. i wonder if there's a sane default heuristic that we can use for guessing which one it probably is?

Comment (masih) [reply cmd/provider/get.go:88]: My suggestion would be to update that in go-legs and reflect here. Cause right now creating a new syncer requires topic: https://github.com/filecoin-project/go-legs/blob/main/dtsync/sync.go#L103

Which I think is used only for head query.

**Review by willscott — APPROVED** [review]

File: cmd/provider/get.go (hunk 2 above)
Comment (willscott) [reply cmd/provider/get.go:88]: fair enough - though i guess the use in diagnostics is different from the main protocol where it should be known in advance

**Pull request merged**